packages/vite-node/src/cli.ts

```diff
@@ -89,16 +89,32 @@ async function run(files: string[], options: CliOptions = {}) {
 }
 
 function parseServerOptions(serverOptions: ViteNodeServerOptionsCLI): ViteNodeServerOptions {
+  const serverOptionsDepsInline = typeof serverOptions.deps?.inline === 'string'
+    ? [serverOptions.deps?.inline]
+    : serverOptions.deps?.inline
+
+  const serverOptionsDepsExternal = typeof serverOptions.deps?.external === 'string'
+    ? [serverOptions.deps?.external]
+    : serverOptions.deps?.external
+
+  const serverOptionsTransformModeSsr = typeof serverOptions.transformMode?.ssr === 'string'
+    ? [serverOptions.transformMode?.ssr]
+    : serverOptions.transformMode?.ssr
+
+  const serverOptionsTransformModeWeb = typeof serverOptions.transformMode?.web === 'string'
+    ? [serverOptions.transformMode?.web]
+    : serverOptions.transformMode?.web
+
   return {
     ...serverOptions,
     deps: {
       ...serverOptions.deps,
-      inline: serverOptions.deps?.inline?.map((dep) => {
+      inline: serverOptionsDepsInline?.map((dep) => {
```

Comment:
You may just use [`toArray`](https://github.com/vitest-dev/vitest/blob/574d072c9250747bd66d09ca1e22df0aeb55681f/packages/vitest/src/utils/base.ts#L84) (you can move it from vitest to vite-node)